lms/static/scripts/frontend_apps/utils/unsaved-changes.ts

```diff
@@ -0,0 +1,46 @@
+let unsavedCount = 0;
```

Comment:
We have variations of this code in three different apps now. _This_ version contains the lowest level API which has no integration with Preact or app services. The `useWarnOnPageUnload` hook in eg. @hypothesis/frontend-shared would need to be built on top of this.